nexus/db-queries/src/db/datastore/bgp.rs

```diff
@@ -41,76 +41,191 @@ impl DataStore {
                             .first_async::<Uuid>(&conn)
-                            .await?
+                            .await
+                            .map_err(|e| {
+                                let msg = "failed to lookup announce set by name";
+                                error!(opctx.log, "{msg}"; "error" => ?e);
+
+                                match e {
+                                    diesel::result::Error::NotFound => {
+                                        err.bail(Error::not_found_by_name(
+                                            ResourceType::BgpAnnounceSet,
+                                            &name,
+                                        ))
+                                    }
+                                    _ => err.bail(Error::internal_error(msg)),
+
+                                }
+                            }),
+
+                        // We cannot assume that the provided UUID is actually real.
+                        // Lookup the parent record by UUID to verify that it is valid.
+                        NameOrId::Id(id) => announce_set_dsl::bgp_announce_set
+                            .filter(bgp_announce_set::time_deleted.is_null())
+                            .filter(bgp_announce_set::id.eq(id))
+                            .select(bgp_announce_set::id)
+                            .limit(1)
+                            .first_async::<Uuid>(&conn)
+                            .await
+                            .map_err(|e| {
+                                let msg = "failed to lookup announce set by id";
+                                error!(opctx.log, "{msg}"; "error" => ?e);
+
+                                match e {
+                                    diesel::result::Error::NotFound => {
+                                        err.bail(Error::not_found_by_id(
+                                            ResourceType::BgpAnnounceSet,
+                                            &id,
+                                        ))
+                                    }
+                                    _ => err.bail(Error::internal_error(msg)),
+
+                                }
+                            }),
+                    }?;
+
+                    let config =
+                        BgpConfig::from_config_create(config, announce_set_id);
+
+                    // TODO: remove once per-switch-multi-asn support is added
+                    // Bail if an existing config for this ASN already exists.
+                    // This is a temporary measure until multi-asn-per-switch is supported.
+                    let configs_with_asn: Vec<BgpConfig> = dsl::bgp_config
+                        .filter(dsl::asn.eq(config.asn))
+                        .filter(dsl::time_deleted.is_null())
+                        .select(BgpConfig::as_select())
+                        .load_async(&conn)
+                        .await?;
```

Comment:
So I found something that needs to be changed here. Elsewhere in Nexus our logic relies on this code being idempotent, so I'm reworking this to allow and exact match to return `Ok(config)` but otherwise bail with the error.

Reply:
This still might not be fixed, doing a bit more testing

Reply:
I think I have it sorted, it was another diesel-ism

Reply:
It's been happy for multiple `a4x2` laps now, auto-merging